src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,293 @@
+use anyhow::anyhow;
+use bigdecimal::BigDecimal;
+use chrono::{DateTime, Duration, Utc};
+use sqlx::{PgConnection, PgPool, Postgres, Transaction};
+
+use super::{Data, DataAggregationError, DataAggregationResult};
+
+pub struct Leaderboards {
+    pool: PgPool,
+    last_indexed_timestamp: Option<DateTime<Utc>>,
+}
+
+impl Leaderboards {
+    pub fn new(pool: PgPool) -> Self {
+        Self {
+            pool,
+            last_indexed_timestamp: None,
+        }
+    }
+}
+
+struct Competition {
+    id: i32,
+    start: DateTime<Utc>,
+    end: DateTime<Utc>,
+    prize: i64,
+    market_id: BigDecimal,
+    integrators_required: Vec<String>,
+}
+
+#[async_trait::async_trait]
+impl Data for Leaderboards {
+    fn model_name(&self) -> &'static str {
+        "Leaderboard"
+    }
+
+    fn ready(&self) -> bool {
+        self.last_indexed_timestamp.is_none()
+            || self.last_indexed_timestamp.unwrap() + Duration::seconds(5) < Utc::now()
```

Comment:
the interval of aggregation should probably be a function on the trait like `get_interval` so that `ready` can have a standardized implementation, and we can avoid _subtle_ bugs where we try to change this interval and forget to change it in other places.

Reply:
I put `TIMEOUT` on both. It's ok not to use `poll_interval` here, since that is just a function for the aggregator to know how much to wait before polling again.